scripts/data/src/org/akvo/gae/remoteapi/ExtractImageGeotag.java

```diff
@@ -0,0 +1,282 @@
+        }
+        //System.out.printf("#S3-bucketname %s, S3-id %s, S3-secret %s\n", S3bucket, S3id, S3secret);
+        processQuestions(ds);
+    }
+
+    private void processQuestions(DatastoreService ds) throws ParseException {
+        System.out.println("#Processing Questions from 2018 and later");
+        int total = 0;
+        int nontagged = 0;
+        int json = 0;
+        int nonjson = 0;
+        int tagged = 0;
+        SimpleDateFormat df = new SimpleDateFormat("YYYY-MM-dd");
+        Date tooOld = df.parse("2018-01-01");
+
+        final Query qq = new Query("QuestionAnswerStore").setFilter(new Query.FilterPredicate("type", FilterOperator.EQUAL, "IMAGE"));
+        final PreparedQuery qpq = ds.prepare(qq);
+        List<Entity> questionsToFix = new ArrayList<Entity>();
+
+        for (Entity q : qpq.asIterable(FetchOptions.Builder.withChunkSize(500))) {
+            total++;
+            boolean forceSave = false;
+            Media media;
+            String v = (String) q.getProperty("value"); //Assume valueText was not used for this image
+            Date collected = (Date) q.getProperty("collectionDate");
+            if (collected.before(tooOld)) continue; //Debug! Concentrate on recent data.
+
+            System.out.printf(" Old IMAGE value '%s'\n", v);
+            if (v != null && !v.trim().equals("")) {
+                if (v.startsWith("{")) {
+                    json++;
+                } else {
+                    nonjson++;
+                    forceSave = true;//handle legacy values: convert them to JSON while we're here
+                    v = Paths.get(v).getFileName().toString(); //strip path, it is never used
+                }
+                String filename;
+                //Parse it
+                media = MediaResponse.parse(v);
+                if (media.getLocation() != null) { //Already known TODO check validity?
+                    //System.out.println("Location in " + q);
+                    tagged++;
+                    continue; //Skip
+                }
+                //also want to skip if location is present, but null, to avoid re-evaluation
+                if (v.matches("\"location\":null")) {
+                    System.out.printf("Null location in IMAGE %d: '%s'\n", q.getKey().getId(), v);
+                    tagged++;
+                    continue; //Skip
+                }
+            } else {
+                System.out.printf("#ERROR null or empty value for IMAGE %d: '%s'\n", q.getKey().getId(), v);
+                continue; //TODO convert this to JSON too?
+            }
+            //No location known; must read the file
+            Location loc = new Location();
+            Boolean tagFound = fetchLocationFromJpegInS3(media.getFilename(), loc);
+            if (tagFound != null || forceSave) {
+
+                if (tagFound == null) { // We cannot know (right now - file may arrive later)
+                    v = MediaResponse.formatWithoutGeotag(media);
```

Comment:
Wouldn't this save the location value as null and the when file arrives it will be skipped?

Reply:
No, this will save it with just the "filename": tag